compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -145,6 +148,27 @@ kindOf = \case
   TStruct recordType -> checkRecordType recordType $> KStar
   TNat _ -> pure KNat
 
+expandTypeSynonyms :: MonadGamma m => Type -> m Type
+expandTypeSynonyms = expand where
+  expand = \case
+    TVar v -> return $ TVar v
+    TSyn tsyn -> do
+      DefTypeSyn{synType} <- inWorld (lookupTypeSyn tsyn)
```

Comment:
Happy to do this in two steps and not handle type synonyms with parameters as part of this PR but then let’s error out if there are parameters instead of silently doing the wrong thing.

Reply:
As discussed, we are adding support for parameterized type synonyms to this PR.